### ISSUE TITLE
Fails to build on Arch Linux x64

### ISSUE THREAD

**Issue by amno1** [lwthiker/curl-impersonate#169]
Hi, I am trying to build the firefox version and I am getting errors building with ssl:
```

make[1]: Lämnar katalogen ”/home/arthur/repos/curl-impersonate/build/brotli-1.0.9/out”
curl -L -o nss-3.87.tar.gz https://ftp.mozilla.org/pub/security/nss/releases/NSS_3_87_RTM/src/nss-3.87-with-nspr-4.35.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 69.6M  100 69.6M    0     0  9398k      0  0:00:07  0:00:07 --:--:-- 10.9M
tar xf nss-3.87.tar.gz
# Native build, use NSS' build script.
cd nss-3.87/nss
./build.sh -o --disable-tests --static --python=python3
# Hack for macOS: Remove dynamic libraries to force the linker to use the
# static ones when linking curl.
rm -Rf /home/arthur/repos/curl-impersonate/build/nss-3.87/dist/Release/lib/*.dylib
NSPR [1/5] configure ...
NSPR [2/5] make ...
NSPR [3/5] NOT building tests
NSPR [4/5] NOT running tests
NSPR [5/5] install ...
ninja: Entering directory `/home/arthur/repos/curl-impersonate/build/nss-3.87/nss/out/Release'
[259/850] CC obj/lib/ssl/ssl.ssl3exthandle.o
FAILED: obj/lib/ssl/ssl.ssl3exthandle.o 
cc -MMD -MF obj/lib/ssl/ssl.ssl3exthandle.o.d -DNSS_FIPS_DISABLED -DNSS_NO_INIT_SUPPORT -DNSS_X86_OR_X64 -DNSS_X64 -DNSS_USE_64 -DNSS_ALLOW_SSLKEYLOGFILE -DUSE_UTIL_DIRECTLY -DNO_NSPR_10_SUPPORT -DSSL_DISABLE_DEPRECATED_CIPHER_SUITE_NAMES -DLINUX2_1 -DLINUX -Dlinux -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -DHAVE_STRERROR -DXP_UNIX -D_REENTRANT -DNSS_DISABLE_DBM -DNSS_DISABLE_LIBPKIX -DNDEBUG -I/home/arthur/repos/curl-impersonate/build/nss-3.87/dist/Release/include/nspr -I/home/arthur/repos/curl-impersonate/build/nss-3.87/dist/private/nss -I/home/arthur/repos/curl-impersonate/build/nss-3.87/dist/public/nss -fPIC -pipe -ffunction-sections -fdata-sections -m64 -Werror -Wall -Wshadow -O2 -std=c99  -c ../../lib/ssl/ssl3exthandle.c -o obj/lib/ssl/ssl.ssl3exthandle.o
../../lib/ssl/ssl3exthandle.c:205:1: fel: motstridiga typer för ”ssl3_ClientSendSessionTicketXtn” på grund av enum/heltal som inte stämmer; har ”PRInt32(const sslSocket *, TLSExtensionData *, sslBuffer *, PRBool *)” {även ”int(const struct sslSocketStr *, struct TLSExtensionDataStr *, struct sslBufferStr *, int *)”} [-Werror=enum-int-mismatch]
  205 | ssl3_ClientSendSessionTicketXtn(const sslSocket *ss, TLSExtensionData *xtnData,
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
I filen inkluderad ifrån ../../lib/ssl/ssl3exthandle.c:17:
../../lib/ssl/ssl3exthandle.h:116:11: anm: tidigare deklaration av ”ssl3_ClientSendSessionTicketXtn” med typen ”SECStatus(const sslSocket *, TLSExtensionData *, sslBuffer *, PRBool *)” {även ”enum _SECStatus(const struct sslSocketStr *, struct TLSExtensionDataStr *, struct sslBufferStr *, int *)”}
  116 | SECStatus ssl3_ClientSendSessionTicketXtn(const sslSocket *ss,
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: alla varningar behandlas som fel
[268/850] CC obj/lib/ssl/ssl.ssl3con.o
ninja: build stopped: subcommand failed.
make: *** [Makefile:211: /home/arthur/repos/curl-impersonate/build/nss-3.87/dist/Release/lib/libnss_static.a] Fel 1
~/repos/curl-impersonate/build main ?
```

Any idea what do I need to do to make it compile?

I have just urn ./configure without any flags, do I need to pass some flag to the configure script?

Running the latest gcc, libs, tools etc. as available on Arch.

**Comment by lwthiker** [comment]
Thanks, this seems to happen because of incompatibility of nss-3.87 with gcc 13. I will upgrade our nss version and see if it fixes it..

**Comment by lwthiker** [comment]
Supposed to be fixed with #176. Please let me know if it the problem repeats.

**Comment by amno1** [comment]
Forgive me for the very slow, or rather lack of the response.; I am just too busy with the life. I am away from home, but I'll try next week and let you know if there is a problem. Thank you very much for your work! Take care.
